src/Analyzers/CSharp/Tests/UseUTF8StringLiteral/UseUTF8StringLiteralTests.cs

```diff
@@ -0,0 +1,1406 @@
+    }
+}",
+                LanguageVersion = LanguageVersion.CSharp10
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWithoutInitializer()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[10];
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotInExpressionTree()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+using System;
+using System.Linq.Expressions;
+
+public class C
+{
+    public void M()
+    {
+        N(() => new byte[] { 65, 66, 67 });
+    }
+
+    public void N(Expression<Func<byte[]>> f)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenNotByteArray()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new int[] { 65, 66, 67 };
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenOptionNotSet()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[] { 65, 66, 67 };
+    }
+}",
+                EditorConfig = @"
+[*.cs]
+csharp_style_prefer_utf8_string_literals = false
+",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenNonLiteralElement()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[] { 65, GetB(), 67 };
+    }
+
+    public byte GetB() => 66;
+}",
+                LanguageVersion = LanguageVersion.Preview
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseUTF8StringLiteral)]
+        public async Task TestNotWhenMultidimensionalArray()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode =
+@"
+public class C
+{
+    public void M()
+    {
+        var x = new byte[,] { { 65, 66 }, { 67, 68 }, { 69, 70 } };
+    }
+}",
+                LanguageVersion = LanguageVersion.Preview
```

Comment:
@jcouv  for guidance.  i think we have a way of sayign "test this in C# 11, even though the compiler hasn't defined that constant yet".

Reply:
For the actual analyzer I am following the expected design. I didn't change the tests because other tests seem to just use Preview